src/agent/src/rpc.rs

```diff
@@ -400,24 +401,45 @@ impl AgentService {
             "exec-id" => eid.clone(),
         );
 
-        let p = sandbox.find_container_process(cid.as_str(), eid.as_str())?;
-
-        let mut signal = Signal::try_from(req.signal as i32).map_err(|e| {
+        let mut sig = Signal::try_from(req.signal as i32).map_err(|e| {
             anyhow!(e).context(format!(
                 "failed to convert {:?} to signal (container-id: {}, exec-id: {})",
                 req.signal, cid, eid
             ))
         })?;
 
+        if eid.is_empty() {
+            // eid is empty, signal all the processes in the container cgroup
+            let ctr = sandbox.get_container(&cid).ok_or_else(|| anyhow!("Invalid container id"))?;
+            match ctr.cgroup_manager {
+                None => Err(anyhow!("container cgroup not exist")),
+                Some(cm) => {
+                    let pids = cm.get_pids();
```

Comment:
CgroupManager.gett_pids calls cgroup-rs' `tasks()` function, which returns values of tasks that may contain threads that are not in the container's process(`sandbox.find_process(pid)`) list. So considering doesn't return an error at line 422?